stuff/cron/assign_badges.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/python3
+
+''' Assigns users badges and creates the notifications.'''
+
+import argparse
+import json
+import logging
+import os
+
+import MySQLdb
+
+import lib.db
+import lib.logs
+
+
+BADGES_PATH = os.path.abspath(os.path.join(__file__, '..', '..',
+                                           '..', 'frontend/badges'))
+
+
+def delete_read_notifications(cur: MySQLdb.cursors.DictCursor):
+    '''Deletes all read notifications'''
+    cur.execute("""
+        DELETE FROM
+            `Notifications`
+        WHERE
+            `read` != 0;""")
+    logging.info('Deleted all already read notifications, %s in total.',
+                 cur.rowcount)
+
+
+def get_all_owners(badge: str, cur: MySQLdb.cursors.DictCursor):
+    '''Returns a set of ids of users who should receive the badge'''
+    with open(os.path.join(BADGES_PATH, badge, 'query.sql')) as fd:
+        query = fd.read()
+    cur.execute(query)
+    results = []
+    for row in cur:
+        results.append(row['user_id'])
+    return set(results)
+
+
+def get_current_owners(badge: str, cur: MySQLdb.cursors.DictCursor):
+    '''Returns a set of ids of current badge owners'''
+    cur.execute("""
+        SELECT
+            ub.user_id
+        FROM
+            Users_Badges ub
+        WHERE
+            ub.badge_alias = '%s';""" % badge)
+    results = []
+    for row in cur:
+        results.append(row['user_id'])
+    return set(results)
+
+
+def save_new_owners(badge: str, users: set, cur: MySQLdb.cursors.DictCursor):
```

Comment:
aquí esta función no regresa nada, pero igual le puedes poner type hints:

```suggestion
def save_new_owners(badge: str, users: set, cur: MySQLdb.cursors.DictCursor) -> None:
```